### ISSUE TITLE
tm2z timeouts

### ISSUE THREAD

**Issue by rclark** [mapbox/mapbox-upload-validate#10]
I encountered a tm2z file that referenced a very massive backend vector tile source. Validation threw the following unexpected error:

```
{ [Error: Timed out after 5000ms] status: 504 }
```

Unexpected means: it did not turn the file away as invalid. Instead it marked the problem as one that is "blippy" and ought to be retried. Subsequent attempts all failed the same way.

The error was likely from one of several `getTile` requests from [tilelive-vector during profiling](https://github.com/mapbox/tilelive-vector/blob/master/index.js#L283). It ends up coming out through [this code in mapbox-upload-validate](https://github.com/mapbox/mapbox-upload-validate/blob/6d2d600a25f48dfc9a3797e93d957a6ec4c67083/lib/validators/tm2z.js#L18).

I have a tm2z file that replicates the problem, except it depends on a vector-tile backend that may be ephemeral. I'm trying to build this into a persistent testcase.

cc @springmeyer

**Comment by GretaCB** [comment]
Just got same error today in unpacker for a different file.

**Comment by rclark** [comment]
I copied the file @GretaCB found to `s3://mapbox/tmp/rclark/fifteen.tm2z`. The only thing that immediately struck me in looking at project.xml is that this is a style that composites 15 sources. My guess is that this increases the the number of times it is possible for a tile request to timeout to the point that each of unpacker's (now 13) attempts to run the file fail.

**Comment by springmeyer** [comment]
its exciting that we might have a testcase that can replicate this.

**Comment by rclark** [comment]
Yes, this timeout is in node.get, where @springmeyer linked above. If I `curl` for one of the vector tiles that times out, it took my machine 18s to download 850K worth of vector tile. The request is set to timeout after 5s.

- there's a limit on the size of individual vector tiles, but is there any limit on the size of 15 composited tiles? Have we thought about how that relates to this 5s timeout?
- should timeouts in tilelive-vector's profile handle these timeouts instead of failing out? This could allow mapbox-upload-validate to mark the style as invalid.

**Comment by springmeyer** [comment]
> it took my machine 18s to download 850K worth of vector tile

Interesting. I presume the majority of time was due to the server working to generate the composited payload.

> is there any limit on the size of 15 composited tiles?

Not that I am aware of.

**Comment by rclark** [comment]
> it took my machine 18s to download 850K worth of vector tile

I misread curl output -- I downloaded 850K/s for a total of 16.5MB worth of vector tile.

**Comment by springmeyer** [comment]
woah, okay. hrmmmmm

**Comment by yhahn** [comment]
chocolate ganache ravioli

**Comment by GretaCB** [comment]
[Related upload with same error](https://gist.github.com/GretaCB/a756f2597fb75e4b95bf) resolved itself, might help define a size/composite threshold:

```
s3://mapbox/tmp/carol/timeout-resolved.tm2z
```
